File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/ItemTransferProperties.java
package com.NovumScientiaTeam.novumscientia.api.materialProperties.transfer;

import com.EmosewaPixel.pixellib.materialsystem.materials.Material;

import java.util.HashMap;
import java.util.Map;

public class ItemTransferProperties extends TransferProperties {
    private static final Map<Material, ItemTransferProperties> itemTransferMap = new HashMap<>();

    public static void setMaterialProperties(Material mat, int itemCapacity, int itemsPerTick) {
        itemTransferMap.put(mat, new ItemTransferProperties(itemCapacity, itemsPerTick));
    }

    public static ItemTransferProperties getPropertiesForMaterial(Material mat) {
        return itemTransferMap.get(mat);
    }

    private ItemTransferProperties(int itemCapacity, int itemsPerTick) {
        super(itemCapacity, itemsPerTick);
    }
}

File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/ItemTransferProperties.kt
package com.novumscientiateam.novumscientia.api.materialproperties.transfer

import com.emosewapixel.pixellib.materialsystem.materials.Material

import java.util.HashMap

class ItemTransferProperties private constructor(itemCapacity: Int, itemsPerTick: Int) : TransferProperties(itemCapacity, itemsPerTick) {
    companion object {
        private val itemTransferMap = HashMap<Material, ItemTransferProperties>()

        fun setMaterialProperties(mat: Material, itemCapacity: Int, itemsPerTick: Int) {
            itemTransferMap[mat] = ItemTransferProperties(itemCapacity, itemsPerTick)
        }

        fun getPropertiesForMaterial(mat: Material) = itemTransferMap[mat]
    }
}